base/asyncmap.jl

```diff
@@ -236,7 +236,7 @@ function start_worker_task!(worker_tasks, exec_func, chnl, batch_size=nothing)
             end
         catch e
             close(chnl)
-            retval = e
+            retval = capture_exception(e, catch_backtrace())
```

Comment:
Could we potentially make use of the exception stack here now?
```suggestion
    catch
        close(chnl)
        retval = Base.ExceptionStack(Any[capture_exception(ex...) for ex in current_exceptions()])
    end
```

Reply:
Thanks for taking a look. I think there should be a way to use ExceptionStacks here but it’s a bit subtle. I think we’d need more changes in Distributed to facilitate this, since code there is expecting RemoteExceptions to be thrown and to handle them specially. The current approach works around that by continuing to throw them.